metadata-ingestion/src/datahub/cli/check_cli.py

```diff
@@ -391,29 +395,78 @@ def test_path_spec(config: str, input: str, path_spec_key: str) -> None:
         raise e
 
 
+def _jsonify(data: Any) -> Any:
+    if dataclasses.is_dataclass(data):
+        # dataclasses.asdict() is recursive, which is not what we want.
```

Comment:
Why don't we want dataclassess recursion if we do as well in recursion?

Reply:
updated the comment to clarify